src/Collection/CollectionInterface.php

```diff
@@ -43,7 +43,7 @@
      * in this collection
      * @return \Cake\Collection\CollectionInterface
      */
-    public function each(callable $c): CollectionInterface;
+    public function each(callable $c);
```

Comment:
Why is this method losing its return type?

Reply:
Could we make `each()` also return new collection in 4.0?

Reply:
sounds like an error in the static analysis. Is it maybe because the TreePrinter is not a CollectionInterface itself?

Reply:
to answer myself: it is not a CollectionInterface It is definitely missing that

Reply:
Indeed, I'll make TreePrinter implement CollectionInterface.